docs/21-interrupts.markdown

```diff
@@ -0,0 +1,697 @@
+
+If `LISTEN` or `ERROR` is called when the `INT-LEVEL` is not zero,
+then the typeout will be
+
+    LISTENING-AT-LEVEL I PROCESS p INT-LEVEL i
+
+### 21.7.3. DISMISS
+
+`DISMISS` permits a handler to return an arbitrary value for an
+arbitrary `ACTIVATION` at an arbitrary interrupt level. The call is as
+follows:
+
+    <DISMISS value:any activation int-level:fix>
+
+where only the *value* is required. If *activation* is omitted, return
+is to the place interrupted from, and *value* is ignored. If
+*int-level* is omitted, the `INT-LEVEL` prior to the current interrupt
+is restored.
+
+## 21.8. Specific Interrupts
+
+Descriptions of the characteristics of particular "built-in" MDL
+interrupts follow. Each is named by its `STRING` name. Expect this
+list to be incomplete yesterday.
+
+`"CHAR"` is currently the most complex built-in interrupt, because it
+serves duty in several ways. These different ways will be described in
+several different sections. All ways are concerned with characters or
+machine words that arrive or depart at unpredictable tunes, because
+MDL is communicating with a person or another processor. Each `"CHAR"`
+`IHEADER` has a `CHANNEL` for the element that names the interrupt,
+and the mode of the `CHANNEL` tells what kinds of `"CHAR"` interrupts
+occur to be handled through that `IHEADER`.
+
+1. If the `CHANNEL` is for `INPUT`, "CHAR" occurs every time an
+"interesting" character (see below) is received from the `CHANNEL`'s
+real terminal, or any character is received from the `CHANNEL`'s
+pseudo-terminal, or a character or word is received from the
+`CHANNEL`'s Network socket or indeed (in the ITS version) the
+operating system generates an interrupt for any reason.
+2. If the `CHANNEL` is for output to a pseudo-terminal or Network
+socket, `"CHAR"` occurs every time a character or word is wanted.
+3. If the `CHANNEL` is for output to a terminal, `"CHAR"` occurs every
+time a line-feed character is output or (in the ITS version) the
+operating system generates a screen-full interrupt for the terminal.
+
+### 21.8.1. "CHAR" received
+
+A handler for an input `"CHAR"` interrupt on a real terminal must take
+two arguments: the `CHARACTER` which was typed, and the `CHANNEL` on
+which it was typed.
+
+In the ITS version, the "interesting" characters are those "enabled
+for interrupts" on a real terminal, namely ^@ through ^G, ^K through
+^_, and DEL (that is, ASCII codes 0-7, 13-37, and 177 octal.)
+
+An initial MDL already has `"CHAR"` enabled on `,INCHAN` with a
+priority 8 (eight), the `SUBR` `QUITTER` for a handler to run in
+`#PROCESS 0` (the running `PROCESS`); this is how `^G` and `^S` are
+processed. In addition, every time a new `CHANNEL` is `OPEN`ed in
+`"READ"` mode to a terminal, a similar `IHEADER` and `HANDLER` are
+associated with that new `CHANNEL` automatically. These
+automatically-generated `IHEADER`s and `HANDLER`s use the standard
+machinery, and they can be `DISABLE`d or `OFF`ed at will. However, the
```

Comment:
bold **However**

(side note: I think `$` in the next line is fine without a kbd tag, since it doesn't represent the dollar sign key)